### ISSUE TITLE
Unable to run simple project from archetype using gradle

### ISSUE THREAD

**Issue by kost-ru** [JacpFX/JacpFX#21]
Hey guys,

I've recently started playing with your framework and as a fan of gradle, I converted maven project into gradle one. After running main class I've got 5-6 ClassNotFoundException.
The problem was with ClassFinder:170. You assumed, that all class-files with package-dirs are in "classes"-folder. But gradle keeps file in a bit different way:
```
MacBook-Pro-Konstantin:perspective konstantinpasko$ ls
PerspectiveOne.class	PerspectiveTwo.class
MacBook-Pro-Konstantin:perspective konstantinpasko$ pwd
/Users/konstantinpasko/test-jacpfx/build/classes/production/test-jacpfx/com/gui/perspective
```
I've fixed in this pull-request (https://github.com/JacpFX/JacpFX/pull/20). I hope it doesn't break anything else.

Regards,
Konstantin.

**Comment by amoAHCP** [comment]
Hi Konstantin,
thank you for your efforts. Can you please provide me a JacpFX gradle example project? I am new to gradle but I would like to test the implementation with maven and gradle… with you push request I’am unsure if it still works in maven. With an example project and your changes I can manage to run JacpFX in both worlds.

Thank you and regards

Andy


> Am 20.01.2015 um 23:43 schrieb kost-ru <notifications@github.com>:
> 
> Hey guys,
> 
> I've recently started playing with your framework and as a fan of gradle, I converted maven project into gradle one. After running main class I've got 5-6 ClassNotFoundException.
> The problem was with ClassFinder:170. You assumed, that all class-files with package-dirs are in "classes"-folder. But gradle keeps file in a bit different way:
> 
> MacBook-Pro-Konstantin:perspective konstantinpasko$ ls
> PerspectiveOne.class    PerspectiveTwo.class
> MacBook-Pro-Konstantin:perspective konstantinpasko$ pwd
> /Users/konstantinpasko/test-jacpfx/build/classes/production/test-jacpfx/com/gui/perspective
> I've fixed in this pull-request (#20 <https://github.com/JacpFX/JacpFX/pull/20>). I hope it doesn't break anything else.
> 
> Regards,
> Konstantin.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/JacpFX/JacpFX/issues/21>.
> 

Andy Moncsek
amo@mailbox.org

**Comment by kost-ru** [comment]
Hi Andy,

sure, I will upload the project to the google.drive then.

Regards,
Konstnatin

**Comment by kost-ru** [comment]
Hi Andy,

I investigated my case more careful:
I use IDEA as IDE and gradle as build system.
IDEA can use gradle’s dependencies and allows to run main-class regardless to maven/gradle goals/tasks.

When I execute main-class (button Run), IDEA compiles all classes to /build/production/%project_name% and then launches main-class.

Here is the simple project (https://drive.google.com/file/d/0B02wOMjV0WoCYms2U1d0SXdfejQ/view?usp=sharing). I’ve just created new project from maven archetype, then converted it into gradle project, then exported it into IDEA. Also I applied gradle-javafx plugin so that it was possible to build runnable jar.

http://www.gradle.org/downloads
https://www.jetbrains.com/idea/download/

Then I installed fixed version of jacpfx.JavaFX into my local maven repository.

The tests cases I checked (all of them use only local repository: flag -o for maven, mavenLocal() for gradle):
1) mvn -o clean package ; java -jar target/test-app.jar — PASS
2) gradle clean assemble ; java -jar build/distributions/test-1.0-SNAPSHOT.jar — PASS
3) gradle clean run — PASS
4) import to IDEA, open main-class, right click, run — PASS

Hope it covers possible ways of running/building.

Regards,
Konstantin
